app/controllers/books_controller.rb

```diff
@@ -69,9 +69,18 @@ def update
 
   def destroy
     @book = Book.find(params[:id])
-    @book.destroy
-    flash[:notice] = 'Book was successfully deleted'
-    redirect_to root_path
+
+    respond_to do |format|
+      format.html do
+        @book.destroy
+        flash[:notice] = 'Book was successfully deleted'
+        redirect_to root_path
+      end
+
+      format.json do
+        @book.destroy
```

Comment:
You could move `@book.destroy` up to above the `respond_to` if you wanted. Then in the `format.json` block you could just respond with:

```
head :ok
```